infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/NamespaceableConfigurationProvisioner.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.provision;
+
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespace;
+
+public interface NamespaceableConfigurationProvisioner<T extends KubernetesEnvironment>
```

Comment:
Pls suggest better name

Reply:
is there a problem to inject KubernetesNamespace to the existed provisioners?

Reply:
yes

Reply:
Let's be honest here, we're not actually using this interface, neither so `ConfigurationProvisioner`. We're only implementing it, but not actually binding to it or using it to polymorphism in other way. So do we need it?

Reply:
not actual anymore